src/baseclasses/chemical_energy/nome_cp_analysis.py

```diff
@@ -0,0 +1,145 @@
+            last_oer_run = self.inputs[-1].reference
+
+            for cp_entry in self.inputs:
+                cp_entry.name = cp_entry.reference.data_file
+                if cp_entry.reference.datetime < first_oer_run.datetime:
+                    first_oer_run = cp_entry.reference
+                if cp_entry.reference.datetime > last_oer_run.datetime:
+                    last_oer_run = cp_entry.reference
+
+            voltage_avg_first5 = np.mean(np.array(first_oer_run.voltage[:5]))
+            voltage_avg_last5 = np.mean(np.array(last_oer_run.voltage[-5:]))
+            voltage_difference = voltage_avg_first5 - voltage_avg_last5
+
+            current_ma = first_oer_run.properties.step_1_current * 1000  # convert A to mA
+            current_density = current_ma / first_oer_run.properties.sample_area
+
+            experiment_duration = 7200  #TODO calculate
+            #experiment_duration = cycle_number * (toer + trecover)
+
+            for sample in first_oer_run.samples:
+                if sample.reference.chemical_composition_or_formulas:
+                    if not archive.results:
+                        archive.results = Results()
+                    if not archive.results.material:
+                        archive.results.material = Material()
+                    try:
+                        from nomad.atomutils import Formula
+                        formula = Formula(sample.reference.chemical_composition_or_formulas)
+                        formula.populate(section=archive.results.material)
+                    except Exception as e:
+                        logger.warn('Could not analyse material', exc_info=e)
+
+            self.outputs = [CPOERAnalysisResult(name=first_oer_run.name,
+                                                voltage_avg_first5=voltage_avg_first5,
+                                                voltage_avg_last5=voltage_avg_last5,
+                                                voltage_difference=voltage_difference,
+                                                j=current_density,
+                                                experiment_duration=experiment_duration,
+                                                reaction_type=first_oer_run.method,
+                                                samples=first_oer_run.samples)]
+            self.outputs[0].normalize(archive, logger)
+        super(CPAnalysis, self).normalize(archive, logger)
+
+
+def find_all_cp_in_upload(data_archive, upload_id):
+    from nomad.search import search
+    from nomad.app.v1.models import MetadataPagination
+
+    # search for all UVvisConcentrationDetection archives
+    query = {
+        'section_defs.definition_qualified_name': 'baseclasses.chemical_energy.chronopotentiometry.Chronopotentiometry',
+        'results.eln.methods': 'OER Chronopotentiometry',
+        'upload_id': upload_id
+    }
+    pagination = MetadataPagination()
+    pagination.page_size = 800
```

Comment:
Is this enough?

Reply:
Don't know. For the summer student the max cycle number was 720. That's why I took 800.

Reply:
Would you take more? How much? 1000?

Reply:
yeah take 10000
i think there is no harm in increasing it this can otherwise lead to weird bugs